### PR TITLE
Add InstantID Pipeline

### PR DESCRIPTION
**Pull request by haofanwang**
# What does this PR do?

This PR adds a community pipeline for our recent [InstantID](https://github.com/InstantID/InstantID) work. The original InstantID pipeline is from [here](https://github.com/InstantID/InstantID/blob/main/pipeline_stable_diffusion_xl_instantid.py), but we add some dependent functions inside the pipeline. We made minimal addition on the top of existing [StableDiffusionXLControlNetPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl.py).

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thank you so much! Could you maybe also add a usage example in the README so that people can use it to the max?

**Comment by haofanwang** [comment]
Done. @sayakpaul

File: examples/community/README.md [hunk 1]
```diff
@@ -3494,3 +3494,65 @@ images = pipeline(
...
+    base_model,
+    controlnet=controlnet,
+    torch_dtype=torch.float16
+).cuda()
+
+# load adapter
+pipe.load_ip_adapter_instantid(face_adapter)
+
+# load an image
+image = load_image("./examples/yann-lecun_resize.jpg")
```
Comment (sayakpaul) [inline examples/community/README.md:3538]: Can we use a URL here?

File: examples/community/README.md [hunk 2]
```diff
@@ -3494,3 +3494,65 @@ images = pipeline(
...
+from insightface.app import FaceAnalysis
+from pipeline_stable_diffusion_xl_instantid import StableDiffusionXLInstantIDPipeline, draw_kps
+
+# prepare 'antelopev2' under ./models
+app = FaceAnalysis(name='antelopev2', root='./', providers=['CUDAExecutionProvider', 'CPUExecutionProvider'])
+app.prepare(ctx_id=0, det_size=(640, 640))
+
+# prepare models under ./checkpoints
+face_adapter = f'./checkpoints/ip-adapter.bin'
+controlnet_path = f'./checkpoints/ControlNetModel'
```
Comment (sayakpaul) [inline examples/community/README.md:3522]: Same for this too.

**Review by sayakpaul — APPROVED** [review]
Excellent! Left some more nits. 

Once addressed, will merge!

**Comment by haofanwang** [comment]
Added. For now, InstantID cannot be loaded via `from_pretrained`, so it is necessary to download beforehand. @sayakpaul

**Comment by sayakpaul** [comment]
> Added. For now, InstantID cannot be loaded via `from_pretrained`, so it is necessary to download beforehand. @sayakpaul

Works. Could you ensure the code quality at least passes?

**Comment by haofanwang** [comment]
Quality check passed.

**Pull request merged**